src/raftstore/store/store.rs

```diff
@@ -232,11 +241,23 @@ impl<T: Transport, C: PdClient> Store<T, C> {
     // may fail, so we allow map_entry.
     #[allow(map_entry)]
     fn on_raft_message(&mut self, mut msg: RaftMessage) -> Result<()> {
+        let region_id = msg.get_region_id();
+
+        let k = if msg.get_message().has_snapshot() {
+            Some(SnapKey::from_region_snap(region_id, msg.get_message().get_snapshot()))
+        } else {
+            None
+        };
+        let mgr = self.snap_mgr.clone();
+        defer!({
+            if let Some(ref k) = k {
+                mgr.wl().deregister(k, false);
```

Comment:
we may return before registering before, then we may deregister the key, no problem?